### ISSUE TITLE
fix(agent): use the witnessed safe Vulkan Qwen Q4_K prefill panel by default

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12745]
## Current disposition — superseded availability failure

Do not land the automatic-128 proposal on the historical failure evidence. Public commit a6ef6c86506b3363fcfe0c46e8ae034b482c192e replaced the portable X-workgroup ceiling with the selected device limit. Newer native 512-token prefill passed on descendant f7f256353f8c4cf179a1abe9ad68a6d7234dad2b. The target device reports X=4,294,967,295; the tested 17,408-by-512 projection needs 139,264 groups. Thus the original HTTP500 is superseded, and this proposed default reduction is not a demonstrated availability repair.

The existing default is retained. The focused software-tested prototype is preserved as an unlanded flow checkpoint for possible measured ablation. No throughput advantage, full-package verification, or hardware qualification is claimed for that prototype. The proposal below is retained as history; any future size tuning must compare matched current-source hardware arms.

# fix(agent): use the witnessed safe Vulkan Qwen Q4_K prefill panel by default

```routing
lane: agent
paths: ["internal/agent/inkernel_planner_config.go", "internal/agent/inkernel_prefill_chunk_test.go", "cmd/fak/serve_qwen_prefill_chunk.go", "cmd/fak/serve_native_config.go", "cmd/fak/serve_qwen_prefill_chunk_test.go"]
expected_steps: 4
```

## Current state

The native serve CLI registers 512 as the Qwen hybrid Q4_K prefill flag default and always copies that value into planner configuration. The planner constructor therefore cannot distinguish an unset automatic setting from an explicit 512-token override. Vulkan Q4_K projections flatten output rows and prompt tokens into the X workgroup dimension. The supported dense model has a 17,408-column FFN projection, so 256- and 512-token panels exceed the portable Vulkan limit of 65,535 X workgroups. The backend rejects the request during tensor preflight, closes the session, and does not retry on CPU.

Source witness: `cmd/fak/serve_qwen_prefill_chunk.go:7-11` defines 512 as the CLI default and validates only 128 through 8,192; `cmd/fak/serve_native_config.go:31-54` registers and copies the value into typed planner configuration; `internal/agent/inkernel_planner_config.go:46-64` resolves that configuration during construction; `internal/compute/vulkan_qwen35_sequence.go:260-268` rejects projections over the portable X limit; `internal/compute/vulkan_shim.cpp:2604-2615` flattens Q4_K projection work as `ceil(out*P/64)` X workgroups; and `internal/compute/qwen35_sequence_contract.go:82-86` fixes the supported dense intermediate width at 17,408.

## Parent context

#8395 tracks useful local agents with accelerated native serving. #11036 owns broad whole-prompt batching across HAL backends. #12001, #12549, and #12697 own separate Vulkan kernel, embedding-copy, and Q8 projection work. None owns the automatic Qwen Q4_K Vulkan chunk selection defect.

## Why now

A physical Radeon 8060S run of Qwen3.8-27B-Q4_K_M witnessed the normal native Vulkan path completing a 2,977-token prompt with 128-token panels, while otherwise matched 256- and 512-token settings both returned HTTP 500 before inference with `projection exceeds portable Vulkan X workgroup limit`. The retained receipt uses schema `fak.strix_halo_benchmark_receipt.v1`, public source `d2e9acb066c06a54a30087ce307a7c0837c3b6e3`, the inkernel engine, and the Vulkan backend. This is a default availability defect on the supported local path, independent of broader kernel tuning.

## Value

Centrality: Core

P1: advanced - normal local Vulkan requests select a physically exercised panel size without operator configuration.
P2: advanced - replace a witnessed HTTP 500 with successful native prefill while preserving model math and generated output; do not claim a new throughput result from the configuration change.
P3: preserved - keep host and Metal effective defaults, explicit overrides, backend validation, fail-closed behavior, and existing sequence capability gates unchanged.
P4: advanced - expose the effective backend-specific automatic default through existing native inference receipts and cover it with deterministic constructor and partition tests.

## Core through-line

Carry zero from the native CLI as the typed `auto` sentinel. During planner construction, resolve that sentinel to 128 tokens only for the stable Vulkan backend identity and to the existing 512-token default for host and Metal paths. Partition divergent Vulkan prompt suffixes through the existing prefill loop, so every automatic panel stays within the physically exercised control. Explicit chunk settings continue unchanged and remain subject to the backend's existing validation and refusal.

## Working spine

Native CLI unset flag -> typed zero `auto` sentinel -> planner constructor chooses Vulkan 128 or existing non-Vulkan 512 -> existing divergent-suffix partition loop -> native Vulkan sequence prefill -> unchanged logits and decode -> existing receipt reports the resolved effective panel size.

## Gold-plating boundary

No shader change, two-dimensional Q4_K dispatch, generalized device-limit query, adaptive autotuning, CPU retry, new public configuration field, or change to host or Metal effective defaults. Do not treat zero as an explicit panel width after construction. Do not clamp or silently rewrite explicit positive overrides. Do not weaken tensor preflight or sequence correctness gates. Do not claim that 128 is universally optimal; it is the bounded physically exercised availability control.

## Done condition

- [ ] [SW-VERIFIED] The native CLI defaults the typed planner field to zero, documents it as automatic, accepts zero during startup validation, and continues rejecting nonzero values outside 128 through 8,192 before model load.
- [ ] [SW-VERIFIED] A constructor receiving the zero sentinel with an eligible Vulkan backend reports and partitions with 128-token prefill panels.
- [ ] [SW-VERIFIED] Constructors receiving the zero sentinel for host and Metal paths retain the 512-token effective default.
- [ ] [SW-VERIFIED] Explicit positive chunk settings retain their exact value on Vulkan and other backends; unsafe Vulkan overrides still reach the existing backend validation and fail closed.
- [ ] [SW-VERIFIED] Existing cancellation boundaries, final-logits-only behavior, non-target routing, and configuration error behavior remain covered by agent package tests.
- [ ] [HW-WITNESSED] The resolving source completes the same supported Vulkan long-prompt command with the automatic setting, records the effective 128-token panel, returns a successful native response, and preserves exact output/quality gates. Throughput is reported as a dated receipt rather than a permanent claim.

## Definition of done

The independent regression shows that the original CLI erases the unset state and passes when zero reaches constructor-level backend selection. Affected `internal/agent` and `cmd/fak` package tests pass, and a source-bound physical rerun records successful automatic native prefill. Host, Metal, and explicit-override controls remain unchanged.

## Acceptance gate

```sh
go test ./internal/agent -run 'TestInKernelQwenQ4KPrefillChunk(Config|ConfigPartitions|BackendDefault)' -count=1
go test ./cmd/fak -run 'TestServeNativeQwenQ4KPrefillChunk(DefaultAndBounds|ReceiptEvidence)' -count=1
go test ./internal/agent ./cmd/fak -count=1
```

Default promotion requires the focused software controls and the physical automatic-setting rerun. The existing 128-token observation is the safe control for implementation; it does not substitute for a resolving-source receipt.

## Closure binding

One verified public commit with `(fak agent)` and the issue reference, independent implementation and test contexts, captured failing-before/passing-after Vulkan default output, affected-package tests, and a source-bound physical automatic-default receipt. A test-only or documentation-only change does not close the issue.

## Witness

```sh
go test ./internal/agent -run TestInKernelQwenQ4KPrefillChunkBackendDefault -count=1
go test ./internal/agent -run TestInKernelQwenQ4KPrefillChunkConfigPartitions -count=1
go test ./cmd/fak -run TestServeNativeQwenQ4KPrefillChunkDefaultAndBounds -count=1
go test ./internal/agent ./cmd/fak -count=1
```

## Verifiable Witness

The pre-fix physical observation used `fak.strix_halo_benchmark_receipt.v1` on public SHA `d2e9acb066c06a54a30087ce307a7c0837c3b6e3`, Qwen3.8-27B-Q4_K_M, inkernel, Vulkan, and a physical Radeon 8060S. The 128-token control completed 2,977 actual prompt tokens. Matched 256- and 512-token arms returned HTTP 500 before inference with the tensor-preflight error `projection exceeds portable Vulkan X workgroup limit`, `session closed, no CPU retry`. Each arm was a single dated observation with high outlier risk; only availability, not a throughput distribution, is claimed. Raw operational receipts remain in the private companion report and are intentionally excluded from the public issue body.

The deterministic regression must pass the CLI's zero sentinel through typed configuration, construct eligible Vulkan, Metal, and host planner controls, distinguish an unset setting from an explicit override, and assert actual prefill call partitions rather than only a helper return value.

## Blast radius and fallback

Only public `cmd/fak` default propagation and `internal/agent` constructor selection for eligible Vulkan Qwen hybrid Q4_K requests change. Existing backend validation remains the fail-closed fallback for explicit unsafe settings. Host, Metal, CPU reference, other model families, other quantizations, and private serving policy are unaffected. No unrelated failing test is disabled.

## Likely files

- `internal/agent/inkernel_planner_config.go`
- `internal/agent/inkernel_prefill_chunk_test.go`
- `cmd/fak/serve_qwen_prefill_chunk.go`
- `cmd/fak/serve_native_config.go`
- `cmd/fak/serve_qwen_prefill_chunk_test.go`

## Lane

agent plus the `cmd/fak` native configuration seam. Coordinates with broad prefill issue #11036 and Vulkan compute issues #12001, #12549, and #12697; no third implementation package or dependency blocks this bounded availability repair.

## Expected steps

Four: freeze independent CLI-sentinel and backend/default/override regressions; carry auto through the CLI and resolve Vulkan-only construction; run focused and affected-package tests; collect the resolving-source physical automatic-default receipt and close only if its availability and quality gates pass.
